src/core/DNft.sol

```diff
@@ -161,30 +160,37 @@ contract DNft is ERC721Enumerable, Owned, IDNft {
     {
       _subDeposit(from, amount); 
       if (_collatRatio(from) < MIN_COLLATERIZATION_RATIO) { revert CrTooLow(); }
-      id2Withdrawn[from] += amount;
+      id2withdrawn[from] += amount;
       dyad.mint(to, amount);
       emit Withdrawn(from, to, amount);
   }
 
   /// @inheritdoc IDNft
   function redeemDyad(uint from, address to, uint amount)
     external 
-      isNftOwnerOrHasPermission(from)
+      isValidNft(from)
       rebase
     returns (uint) { 
-      id2Withdrawn[from] -= amount;
-      dyad.burn(msg.sender, amount); 
+      _burnDyad(from, amount);
       return _redeem(to, amount);
   }
 
+  function _burnDyad(uint from, uint amount)
+    private 
+    returns (uint) {
+      id2withdrawn[from] -= amount;
+      dyad.burn(msg.sender, amount); 
+  }
+
   /// @inheritdoc IDNft
   function redeemDeposit(uint from, address to, uint amount)
     external 
       isNftOwnerOrHasPermission(from)
       rebase
     returns (uint) { 
       _subDeposit(from, amount); 
-      if (_collatRatio(from) < MIN_COLLATERIZATION_RATIO) { revert CrTooLow(); }
+      if (_shares2deposit(id2shares[from]) < MIN_DYAD_DEPOSIT) revert DepositTooLow();
```

Comment:
If this is only checked in redeemDeposit, then we can withdraw it all into DYAD and then redeemDYAD. 

Also, if we're only checking id2shares, then we end up not counting the withdrawn DYAD, which I think we should.

My solution would be:
- The check should be `_shares2deposit(id2shares[from]) + id2withdrawn[from] < MIN_DYAD_DEPOSIT`
- The check should be in both redeemDeposit and redeemDyad

The result should be that we look for the total shares + withdrawn for any DNft, make sure it's sufficiently high (ie they haven't taken too much "out of the system") and that we don't let them take anything out of the system unless that's high enough.